c10/core/TensorImpl.h

```diff
@@ -891,15 +879,18 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
    */
   DeviceType device_type() const {
```

Comment:
Why don't you call `device()` here? You'd get a better error message in that case, no?

Reply:
not a bad idea...I was looking into if we could just kill this (TensorImpl should be minimal, this doesn't add any functionality), but didn't follow through.